diagnosticsModule/Public/Join-DiagnosticsFiles.ps1

```diff
@@ -0,0 +1,106 @@
+            $warningTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select WarningTests
+            $failedTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select FailedTests
+            $errorTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select ErrorTests
+            $notRunTests = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select NotRunTests
+            $reachableServers = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select ReachableServers
+            $unreachableServers = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Test-AdfsServerHealth | Select UnreachableServers
+
+            $mergedTestData["AllTests"] += $allTests.AllTests
+            $mergedTestData["PassedTests"] += $passedTests.PassedTests
+            $mergedTestData["WarningTests"] += $warningTests.WarningTests
+            $mergedTestData["FailedTests"] += $failedTests.FailedTests
+            $mergedTestData["ErrorTests"] += $errorTests.ErrorTests
+            $mergedTestData["NotRunTests"] += $notRunTests.NotRunTests
+            $mergedTestData["ReachableServers"] += $reachableServers.ReachableServers
+            $mergedTestData["UnreachableServers"] += $unreachableServers.UnreachableServers
+
+            $configuration = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object -ExpandProperty ADFSToolbox | Select-Object -ExpandProperty Adfs-Configuration
+
+            # Only add the ADFS Configuration information if this is coming from the primary node
+            if (($configuration | Select-Object -ExpandProperty Role).ToString() -eq "1") {
+                $adfsConfiguration = $configuration
+            }
+
+            $version = @(Get-Content $fileName -raw) | ConvertFrom-Json | Select-Object Version | Select Version
+        }
+        
+    $outputPath = $FilePath + $OutputFileName
+    Write-Host
+    Write-Host "Creating merged diagnostics file: $outputPath"
+
+    $resultantData = @{}
+    $resultantData["Test-AdfsServerHealth"] = @()
+    $resultantData["Test-AdfsServerHealth"] = $mergedTestData
+    $resultantData["Adfs-Configuration"] = $adfsConfiguration
+
+    $mergedOutput = @{}
+    $mergedOutput["ADFSToolbox"] = $resultantData
+    $mergedOutput["Version"] = $version.Version
+
+    $mergedOutput | ConvertTo-JSON -depth 100 -Compress | Out-File $OutputPath -Encoding default
```

Comment:
can you do some research about the depth parameter, I remember that having a high depth is computationally intensive

Reply:
I don't know why this was set to 100 initially.  But i've updated it to set to the same depth as we allow for diagnostics output.